### PR TITLE
Adds testing infrastructure to the common LMLayer project.

### PR DESCRIPTION
**Pull request by jahorton**
While this PR doesn't add any notable test cases, it does provide a couple of token cases to serve as example placement for any developed test cases we may add in the future.  I've largely modeled the infrastructure off of what we already use for KMW, though with an additional section for DOMless testing, as the LMLayer should be able to support that.

A minor guide to interesting files:
* `unit_tests/in_browser/base.conf.js` specifies our base, common configuration for Karma (in-browser) testing.  Two separate *.conf.js files extend this for local vs remote (BrowserStack) testing, the latter of which is very useful for CI integration.
* `unit_tests/test.sh` is the core testing script.  As may be discerned from package.json, calling `npm test` runs this script.  This script in turn runs the DOMless tests first, then calls a separate testing script (`unit_tests/in_browser/browser-test.sh`) for browser-based testing.  A few more CLI options should probably be added for better control of the second script, but the core is ready now.

### CODE REVIEW

File: common/predictive-text/package.json [hunk 1]
```diff
@@ -19,6 +21,20 @@
...
+    "karma-edge-launcher": "^0.4.2",
+    "karma-firefox-launcher": "^1.1.0",
+    "karma-fixture": "^0.2.6",
+    "karma-html2js-preprocessor": "^1.1.0",
+    "karma-ie-launcher": "^1.0.0",
+    "karma-json-fixtures-preprocessor": "0.0.6",
+    "karma-mocha": "^1.3.0",
+    "karma-mocha-reporter": "^2.2.5",
+    "karma-safari-launcher": "^1.0.0",
+    "karma-teamcity-reporter": "^1.1.0",
```
Comment (eddieantonio) [inline common/predictive-text/package.json]: Frankly, I'm really concerned about having two different node environments: one in `common/predictive-text`, and one in `web/`. I think in the future, we may want to look into how these two can share one environment, with the _exact_ same dependencies.

File: common/predictive-text/unit_tests/in_browser/CI.conf.js [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+   * Definition of utility functions for managing our browser lists.
+   */
+  var mergeLaunchers = function() {
+    var mergedDefs = {};
+    var i;
+    for(i=0; i < arguments.length; i++) {
+      for(var name in arguments[i]) {
+        mergedDefs[name] = arguments[i][name];
+
+        // Necessary for Karma to process it properly.
```
Comment (eddieantonio) [inline common/predictive-text/unit_tests/in_browser/CI.conf.js]: Could you clarify what "it" is in this comment?

File: common/predictive-text/unit_tests/in_browser/CI.conf.js [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+      os: 'ios',
+      os_version: '11.0'
+    },
+    bs_ipad6: {
+      device: 'iPad 5th',
+      real_mobile: true,
+      os: 'ios',
+      os_version: '11.0'
+    }
+  };*/
```
Comment (eddieantonio) [inline common/predictive-text/unit_tests/in_browser/CI.conf.js]: In general, I tend to delete entirely commented-out code when using a VCS. Leave the comment, however.

File: common/predictive-text/unit_tests/in_browser/base.conf.js [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+  // frameworks to use
+  // available frameworks: https://npmjs.org/browse/keyword/karma-adapter
+  frameworks: ['mocha', 'chai', 'fixture'],
+
+  // list of files / patterns to load in the browser
+  files: [
+    'unit_tests/in_browser/cases/**/*.js'//, // Where the tests actually reside.
+    //'unit_tests/in_browser/json/**/*.json', // Where pre-loaded JSON resides.
+    //{pattern: 'unit_tests/in_browser/resources/**/*.*', watched: true, served: true, included: false}, // General testing resources.
+    //{pattern: 'unit_tests/in_browser/fixtures/**/*.html', watched: true} // HTML structures useful for testing.
```
Comment (eddieantonio) [inline common/predictive-text/unit_tests/in_browser/base.conf.js]: Delete commented-out code.

File: common/predictive-text/unit_tests/in_browser/CI.conf.js (hunk 2 above)
Comment (jahorton) [reply common/predictive-text/unit_tests/in_browser/CI.conf.js]: Referring to https://github.com/karma-runner/karma-browserstack-launcher, any browser-based tests we want to launch through BrowserStack require having this property set.  There's a 'base' BrowserStack configuration that `karma-browserstack-launcher` provides, and all browser specifications in this file derive from that.

**Comment by eddieantonio** [comment]
Thanks! That that clears up a lot. Merge at will! :shipit:

**Pull request merged**